db/types.go

```diff
@@ -320,3 +324,173 @@ type ProcessesBundle struct {
+		orgAddress: orgAddress,
+		isHashed:   true,
+	}
+}
+
+// String returns a masked version of the phone for display
+func (p *Phone) String() string {
+	if p == nil {
+		return ""
+	}
+
+	if p.original != "" && !p.isHashed {
+		return p.original // Return original if we have it
+	}
+
+	if len(p.hashed) > 0 {
+		// Return last 6 characters of hex representation for display
+		hexHash := fmt.Sprintf("%x", p.hashed)
+		if len(hexHash) >= 6 {
+			return "***" + hexHash[len(hexHash)-6:]
+		}
+		return "***" + hexHash
+	}
+
+	return ""
+}
+
+// GetOriginal returns the original phone number if available
+// This should only be used for 2FA purposes
+func (p *Phone) GetOriginal() string {
+	if p == nil {
+		return ""
+	}
+	return p.original
+}
+
+// GetHashed returns the hashed phone bytes
+func (p *Phone) GetHashed() []byte {
+	if p == nil {
+		return nil
+	}
+	return p.hashed
+}
+
+// IsEmpty returns true if the phone is empty
+func (p *Phone) IsEmpty() bool {
+	if p == nil {
+		return true
+	}
+	return len(p.hashed) == 0 && p.original == ""
+}
+
+// Matches checks if a given phone number matches this hashed phone
+func (p *Phone) Matches(phoneNumber string) error {
+	if p == nil || p.IsEmpty() {
+		return fmt.Errorf("nil phone")
+	}
+
+	normalized, err := internal.SanitizeAndVerifyPhoneNumber(phoneNumber)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(p.hashed, internal.HashOrgData(p.orgAddress, normalized)) {
+		return fmt.Errorf("phone doesn't match stored hash")
+	}
+
+	return nil
+}
+
+// Validate checks if a given phone number is valid
+func (p *Phone) Validate() error {
+	_, err := internal.SanitizeAndVerifyPhoneNumber(p.original)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// MarshalBSONValue implements the bson.ValueMarshaler interface
+func (p *Phone) MarshalBSONValue() (bsontype.Type, []byte, error) {
+	if p == nil || p.IsEmpty() {
+		return bson.TypeNull, nil, nil
+	}
+	// Store as binary data in the database
+	return bson.MarshalValue(p.hashed)
+}
+
+// UnmarshalBSONValue implements the bson.ValueUnmarshaler interface
+func (p *Phone) UnmarshalBSONValue(t bsontype.Type, data []byte) error {
+	if t == bson.TypeNull {
+		*p = Phone{}
+		return nil
+	}
+
+	var hashedPhone []byte
+	if err := bson.UnmarshalValue(t, data, &hashedPhone); err != nil {
+		return err
+	}
+
+	p.hashed = hashedPhone
+	p.isHashed = true
+	return nil
+}
+
+// MarshalJSON implements the json.Marshaler interface
+func (p *Phone) MarshalJSON() ([]byte, error) {
+	// Return the display version for API responses
+	return json.Marshal(p.String())
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface
+func (p *Phone) UnmarshalJSON(data []byte) error {
+	var phoneStr string
+	if err := json.Unmarshal(data, &phoneStr); err != nil {
+		return err
+	}
+
+	p.original = phoneStr
+	p.isHashed = false
+	return nil
+}
+
+// SetOrgAddress sets the organization address for hashing operations
+func (p *Phone) SetOrgAddress(orgAddress common.Address) {
```

Comment:
The naming of the function seems a bit weird since the result could also be the hashing of the data